packages/network-controller/src/NetworkController.ts

```diff
@@ -238,45 +247,44 @@ export class NetworkController extends BaseControllerV2<
     );
   }
 
-  private setupStandardProvider(
-    rpcTarget: string,
-    chainId?: string,
-    ticker?: string,
-    nickname?: string,
+  // extensions network controller saves a copy of the blockTracker.
+  // need to figure out if we migrate this functionality or not.
+  private updateProvider(
+    networkMiddleware: JsonRpcMiddleware<unknown, unknown>,
+    blockTracker: PollingBlockTracker,
   ) {
-    const config = {
-      ...this.internalProviderConfig,
-      ...{
-        chainId,
-        engineParams: { pollingInterval: 12000 },
-        nickname,
-        rpcUrl: rpcTarget,
-        ticker,
-      },
-    };
-    this.updateProvider(createMetamaskProvider(config));
-  }
-
-  private updateProvider(provider: any) {
     this.safelyStopProvider(this.provider);
+
+    const engine = new JsonRpcEngine();
+    engine.push(networkMiddleware);
+    const provider = providerFromEngine(engine);
+
     this.provider = provider;
+    this.blockTracker = blockTracker;
     this.registerProvider();
   }
 
-  private safelyStopProvider(provider: any) {
+  private safelyStopProvider(provider: SafeEventEmitterProvider | undefined) {
+    if (provider === undefined) {
+      return;
+    }
+
     setTimeout(() => {
-      provider?.stop();
+      provider?.removeAllListeners();
```

Comment:
provider.stop doesn't exist anymore - open to suggestions over what we should do here.